components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AbstractRulesValidator.java

```diff
@@ -0,0 +1,145 @@
+    }
+
+    @Override
+    public boolean validate(ValidationContext context) throws InputValidationMgtClientException {
+
+        return true;
+    }
+
+    @Override
+    public boolean validateProps(ValidationContext context) throws InputValidationMgtClientException {
+
+        Map<String, String> properties = context.getProperties();
+        // Validated supported properties.
+        validatePropertyName(properties, this.getClass().getSimpleName(), context.getTenantDomain());
+
+        if (properties.get(MIN_LENGTH) != null && validatePositiveNumber(properties.get(MIN_LENGTH), MIN_LENGTH,
+                context.getTenantDomain())) {
+            properties.remove(MIN_LENGTH);
+        }
+        if (properties.get(MAX_LENGTH) != null && validatePositiveNumber(properties.get(MAX_LENGTH), MAX_LENGTH,
+                context.getTenantDomain())) {
+            properties.remove(MIN_LENGTH);
+        }
+        // Ensure maximum limit is not less than minimum limit.
+        if ((properties.get(MAX_LENGTH) != null) && (properties.get(MIN_LENGTH) != null) &&
+                Integer.parseInt(properties.get(MIN_LENGTH)) > Integer.parseInt(properties.get(MAX_LENGTH))) {
+            if (log.isDebugEnabled()) {
+                log.debug(String.format("The min limit cannot be greater than max limit for %s in tenant: %s",
+                        this.getClass().getSimpleName(), context.getTenantDomain()));
+            }
+            throw new InputValidationMgtClientException(ERROR_DEFAULT_MIN_MAX_MISMATCH.getCode(),
+                    String.format(ERROR_DEFAULT_MIN_MAX_MISMATCH.getDescription(), this.getClass().getSimpleName(),
+                            properties.get(MIN_LENGTH), properties.get(MAX_LENGTH)));
+        }
+        return true;
+    }
+
+
+    @Override
+    public List<Property> getConfigurationProperties() {
+
+        List<Property> configProperties = new ArrayList<>();
+        int parameterCount = 0;
+
+        Property minLength = new Property();
+        minLength.setName(MIN_LENGTH);
+        minLength.setDisplayName("Minimum limit");
+        minLength.setDescription("The minimum length of the field value.");
+        minLength.setType("int");
+        minLength.setDisplayOrder(++parameterCount);
+        configProperties.add(minLength);
+
+        Property maxLength = new Property();
+        maxLength.setName(MAX_LENGTH);
+        maxLength.setDisplayName("Maximum limit");
+        maxLength.setDescription("The maximum length of the field value.");
+        maxLength.setType("int");
+        maxLength.setDisplayOrder(++parameterCount);
+        configProperties.add(maxLength);
+
+        return configProperties;
+    }
+
+    protected boolean validatePositiveNumber(String value, String property, String tenantDomain)
+            throws InputValidationMgtClientException {
+
+        if (!NumberUtils.isNumber(value)) {
```

Comment:
According to this method usage it is for validating whether a string is a positive number. `NumberUtils.isNumber(value)` give `true` negative numbers in string format ("-4").